scholly/base/src/index.js

```diff
@@ -7,11 +7,69 @@ import 'bootstrap/dist/css/bootstrap.css';
 import 'bootstrap/dist/css/bootstrap-grid.css';
 import 'bootstrap/dist/css/bootstrap-reboot.css';
 
-import App from './components/App';
+import { LAYOUT_OPTION } from './global/data';
+
+/// Decides which of the root App components of the various layouts to display.
+/// This determination is based off the provided LAYOUT_OPTION environment variable.
+const layoutToDisplay = () => {
+  switch (LAYOUT_OPTION) {
+    case 1: // TODO: there might be a better way to do this.
+      // eslint-disable-next-line no-case-declarations
+      const AppLayout1 = React.lazy(() => import('./layouts/layout1/App'));
+      return (
+        <>
+          <React.Suspense fallback={<></>}>
+            <AppLayout1 />
+          </React.Suspense>
+        </>
+      );
```

Comment:
The lazy loading is used to optimise production builds. Multiple layout imports cause multiple Layout CSS file imports from each layout's respective App.js. As there may be overlapping class names for components in each layout, the best option is to go for a dynamic import. This will also help in reducing the size of builds, which can be useful when many more layouts are added in the future.

Reply:
@melvinalex7  makes sense.
 
@kdav108 could you investigate this issue and, if we end-up going with lazy loading, please document the reason we're doing so somewhere in this file, perhaps in the file headers. It's not clear upon inspection why we're doing this. Further, the solutions used here are experimental: https://reactjs.org/docs/concurrent-mode-suspense.html, so I think we should try to avoid this if we can.